### DISCUSSION TITLE
변수 스코프란 무엇인가요?

### DISCUSSION THREAD

**Discussion by publdaze** [publdaze/js-study#22] [변수]
- 변수 스코프란 무엇인가요?

**Comment by pipisebastian** [comment]
> 변수 스코프란 무엇인가요?

변수의 유효범위입니다. 코드상에서 변수를 참조해야할때, 현재 스코프를 시작해서 상위 스코프로 계속해서 찾아내야하는데요, 이를 스코프체인이라고 합니다.
스코프체인이 동작하는 원리는 실행컨텍스트의 `outerEnvironment Record` 와 관련이 있습니다. 먼저 자신의 `environment Record` 상에서 찾지 못할 경우, `outerEnvironment Record` 를 통해, 외부 컨텍스트 정보, 즉 상위 스코프로 이동하고, 해당 스코프의 `environment Record` 에서 다시 찾고, 거기서도 없을경우 `outerEnvironment Record` 를 통해, 또다시 상위 스코프로 이동하게 됩니다

**Comment by publdaze** [comment]
> - 변수 스코프란 무엇인가요?
- 변수가 접근할 수 있는 범위
- 변수의 생명주기 결정
- 전역 스코프와 지역 스코프로 나뉘고, 지역 스코프는 함수 스코프와 블록 스코프로 나뉨
